spec/marten/middleware/asset_serving_spec.cr

```diff
@@ -280,5 +280,27 @@ describe Marten::Middleware::AssetServing do
         response.headers[:"Cache-Control"].should eq "private, max-age=3600"
       end
     end
+    it "properly sets the custom Cache-Control header" do
+      Marten.settings.assets.max_age = 7200
+      middleware = Marten::Middleware::AssetServing.new
+
+      ["BigApp.js", "css/BigApp.css"].each do |path|
+        response = middleware.call(
+          Marten::HTTP::Request.new(
+            ::HTTP::Request.new(
+              method: "GET",
+              resource: "/assets/#{path}",
+              headers: HTTP::Headers{
+                "Host"            => "example.com",
+                "Accept-Encoding" => "gzip, deflate, br",
+              }
+            )
+          ),
+          ->{ Marten::HTTP::Response.new("Unknown!", content_type: "text/plain", status: 200) }
+        )
+
+        response.headers[:"Cache-Control"].should eq "private, max-age=7200"
```

Comment:
We should avoid mutating settings in specs. You can make use of the `#with_overridden_setting` method to safely mutate setting values for a specific block instead:

```suggestion
      with_overridden_setting("assets.max_age", 7200) do
        middleware = Marten::Middleware::AssetServing.new
  
        ["BigApp.js", "css/BigApp.css"].each do |path|
          response = middleware.call(
            Marten::HTTP::Request.new(
              ::HTTP::Request.new(
                method: "GET",
                resource: "/assets/#{path}",
                headers: HTTP::Headers{
                  "Host"            => "example.com",
                  "Accept-Encoding" => "gzip, deflate, br",
                }
              )
            ),
            ->{ Marten::HTTP::Response.new("Unknown!", content_type: "text/plain", status: 200) }
          )
  
          response.headers[:"Cache-Control"].should eq "private, max-age=7200"
        end
```

Reply:
updated